Add plugin: AI LLM

Sparky4567 opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

<!--- Paste a link to your repo here for easy access -->
Link to my plugin: https://github.com/Sparky4567/obsidian_ai_plugin

## Release Checklist
- [ + ] I have tested the plugin on
  - [ ]  Windows
  - [ ]  macOS
  - [ + ]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [ ] My GitHub release contains all required files
  - [ + ] `main.js`
  - [ + ] `manifest.json`
  - [ ] `styles.css` _(optional)_
- [ + ] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [ + ] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [ + ] My README.md describes the plugin's purpose and provides clear usage instructions.
- [ + ] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [ + ] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [ + ] I have added a license in the LICENSE file.
- [ + ] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

Sparky4567 commented:
Add plugin: AI LLM

Sparky4567 commented:
All checks passed

joethei commented:
What sets your plugin apart from the other Ollama plugins?

["minAppVersion": "0.15.0",](https://github.com/Sparky4567/obsidian_ai_plugin/blob/6babf091ea1e00469a5acd0cadd6372df9c82ea1/manifest.json#L5)
You are using newer API's, set this to the latest public build number.

["description": "Lets to integrrate local llm (ollama) functionality into your app.",](https://github.com/Sparky4567/obsidian_ai_plugin/blob/6babf091ea1e00469a5acd0cadd6372df9c82ea1/manifest.json#L6)
Typo in `integrrate`

[// Remember to rename these classes and interfaces!](https://github.com/Sparky4567/obsidian_ai_plugin/blob/6babf091ea1e00469a5acd0cadd6372df9c82ea1/main.ts#L12)
Remove this comment

[// This creates an icon in the left ribbon.](https://github.com/Sparky4567/obsidian_ai_plugin/blob/6babf091ea1e00469a5acd0cadd6372df9c82ea1/main.ts#L30-L64)
Remove everything from the sample plugin that you don't need.

Sparky4567 commented:
Good evening, Johannes,

The typo was fixed and unnecessary comments were removed.

My plugin is different because it does not require any other local
installations and downloads except ollama itself. Moreover, it does not
require you to subscribe to any other services to get API keys. It just
works out of the box, if you have models which are specified in the
settings tab.

On Wed, Mar 20, 2024 at 9:23 PM Johannes Theiner ***@***.***>
wrote:

> What sets your plugin apart from the other Ollama plugins?
>
> "minAppVersion": "0.15.0",
> <https://github.com/Sparky4567/obsidian_ai_plugin/blob/6babf091ea1e00469a5acd0cadd6372df9c82ea1/manifest.json#L5>
> You are using newer API's, set this to the latest public build number.
>
> "description": "Lets to integrrate local llm (ollama) functionality into
> your app.",
> <https://github.com/Sparky4567/obsidian_ai_plugin/blob/6babf091ea1e00469a5acd0cadd6372df9c82ea1/manifest.json#L6>
> Typo in integrrate
>
> // Remember to rename these classes and interfaces!
> <https://github.com/Sparky4567/obsidian_ai_plugin/blob/6babf091ea1e00469a5acd0cadd6372df9c82ea1/main.ts#L12>
> Remove this comment
>
> // This creates an icon in the left ribbon.
> <https://github.com/Sparky4567/obsidian_ai_plugin/blob/6babf091ea1e00469a5acd0cadd6372df9c82ea1/main.ts#L30-L64>
> Remove everything from the sample plugin that you don't need.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/obsidianmd/obsidian-releases/pull/3244#issuecomment-2010440291>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHPEC7GVGSAPOGL4YTPBUVTYZHO45AVCNFSM6AAAAABE36L536VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDAMJQGQ2DAMRZGE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Pagarbiai,
Andrius Pratusis

Mobilus: 867640723

joethei commented:
At least two of the plugins that show up when you search for "Ollama" do the same thing, no?

Sparky4567 commented:
One of them:

- Summarizes selection
- Explains selection
- Expands selection
- Rewrites selection (formal)
- Rewrites selection (casual)
- Rewrites selection (active voice)
- Rewrites selection (bullet points)
- Captions selection

Another one:

Requires to run a lightweight python server

My plugin:
- Does not require any additional installations
- Accepts free form written requests (for example, "give me a few ideas for
today's blog post") and it will return the answer

On Wed, Mar 20, 2024 at 9:52 PM Johannes Theiner ***@***.***>
wrote:

> At least two of the plugins that show up when you search for "Ollama" do
> the same thing, no?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/obsidianmd/obsidian-releases/pull/3244#issuecomment-2010501687>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHPEC7FHAT4ZAEFZF4E3KATYZHSHJAVCNFSM6AAAAABE36L536VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDAMJQGUYDCNRYG4>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Pagarbiai,
Andrius Pratusis

Mobilus: 867640723

joethei commented:
["description": "Lets to integrate local llm (ollama) functionality into your app.",](https://github.com/Sparky4567/obsidian_ai_plugin/blob/50fbfed0a18b33f9728c9b3c99c6f7a778673113/manifest.json#L6)
This description is grammatically wrong, and you should avoid sentences like "lets you integrate" or "into the app".
We have published a guide for plugin descriptions to follow here: https://docs.obsidian.md/Plugins/Releasing/Submission+requirements+for+plugins#Keep+plugin+descriptions+short+and+simple.

[this.addCommand({](https://github.com/Sparky4567/obsidian_ai_plugin/blob/50fbfed0a18b33f9728c9b3c99c6f7a778673113/main.ts#L28-L37)
This command makes no sense in this plugin, remove it.

[name: "ASK LLM",](https://github.com/Sparky4567/obsidian_ai_plugin/blob/50fbfed0a18b33f9728c9b3c99c6f7a778673113/main.ts#L96)
Remove the caps from "ASK" here.

[name: "Continue my story and make it better (ASK LLM)",](https://github.com/Sparky4567/obsidian_ai_plugin/blob/50fbfed0a18b33f9728c9b3c99c6f7a778673113/main.ts#L109), [name: "Make a story from my text (ASK LLM)",](https://github.com/Sparky4567/obsidian_ai_plugin/blob/50fbfed0a18b33f9728c9b3c99c6f7a778673113/main.ts#L123), [name: "Summarize my text (ASK LLM)",](https://github.com/Sparky4567/obsidian_ai_plugin/blob/50fbfed0a18b33f9728c9b3c99c6f7a778673113/main.ts#L137)
Including "(ASK LLM)" in the command name is unnecessary, remove it.

Sparky4567 commented:
Dear Johannes Theiner,

Thank you for your email and for providing feedback on my plugin's
description and command structure. I've carefully reviewed your suggestions
and have implemented the necessary changes to ensure compliance with the
guidelines provided.

Here are the adjustments I've made:
Plugin Description:

   - Revised the description to ensure grammatical correctness and
   adherence to best practices. The updated description now aligns with the
   guidelines outlined in your provided documentation.

Command Adjustments:

   - Removed the command that was deemed nonsensical in the context of the
   plugin.
   - Corrected the formatting by removing unnecessary capitalization from
   the command name.
   - Eliminated the redundant inclusion of "(ASK LLM)" in the command names
   as advised.

I appreciate your guidance in maintaining clarity and simplicity in plugin
descriptions and adherence to standardized naming conventions for commands.
Your suggestions have been invaluable in refining my plugin for better
usability and consistency.

Should you have any further recommendations or require additional
information, please don't hesitate to reach out. I'm committed to ensuring
the quality and compliance of my plugins with the established standards.

Thank you once again for your attention to detail and for contributing to
the improvement of my plugin.

Best regards,

Andrius Pratusis



On Thu, Mar 21, 2024 at 1:49 PM Johannes Theiner ***@***.***>
wrote:

> "description": "Lets to integrate local llm (ollama) functionality into
> your app.",
> <https://github.com/Sparky4567/obsidian_ai_plugin/blob/50fbfed0a18b33f9728c9b3c99c6f7a778673113/manifest.json#L6>
> This description is grammatically wrong, and you should avoid sentences
> like "lets you integrate" or "into the app".
> We have published a guide for plugin descriptions to follow here:
> https://docs.obsidian.md/Plugins/Releasing/Submission+requirements+for+plugins#Keep+plugin+descriptions+short+and+simple
> .
>
> this.addCommand({
> <https://github.com/Sparky4567/obsidian_ai_plugin/blob/50fbfed0a18b33f9728c9b3c99c6f7a778673113/main.ts#L28-L37>
> This command makes no sense in this plugin, remove it.
>
> name: "ASK LLM",
> <https://github.com/Sparky4567/obsidian_ai_plugin/blob/50fbfed0a18b33f9728c9b3c99c6f7a778673113/main.ts#L96>
> Remove the caps from "ASK" here.
>
> name: "Continue my story and make it better (ASK LLM)",
> <https://github.com/Sparky4567/obsidian_ai_plugin/blob/50fbfed0a18b33f9728c9b3c99c6f7a778673113/main.ts#L109>,
> name: "Make a story from my text (ASK LLM)",
> <https://github.com/Sparky4567/obsidian_ai_plugin/blob/50fbfed0a18b33f9728c9b3c99c6f7a778673113/main.ts#L123>,
> name: "Summarize my text (ASK LLM)",
> <https://github.com/Sparky4567/obsidian_ai_plugin/blob/50fbfed0a18b33f9728c9b3c99c6f7a778673113/main.ts#L137>
> Including "(ASK LLM)" in the command name is unnecessary, remove it.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/obsidianmd/obsidian-releases/pull/3244#issuecomment-2012059749>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHPEC7EVCKBIFPT2FK5QFMDYZLCLRAVCNFSM6AAAAABE36L536VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDAMJSGA2TSNZUHE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Pagarbiai,
Andrius Pratusis

Mobilus: 867640723
